test/brsTypes/components/RoDouble.test.js

```diff
@@ -0,0 +1,84 @@
+const brs = require("brs");
+const { roDouble, Double, BrsBoolean, BrsString, Callable } = brs.types;
+const { Interpreter } = require("../../../lib/interpreter");
+
+describe("roDouble", () => {
+    describe("equality", () => {
+        it("compares to intrinsic Double", () => {
+            let a = new roDouble(new Double(10.0));
+            let b = new roDouble(new Double(5.5));
+            let c = new roDouble(new Double(123.99));
+            let d = new roDouble(new Double(10.0));
+
+            expect(a.equalTo(b)).toBe(BrsBoolean.False);
+            expect(a.equalTo(c)).toBe(BrsBoolean.False);
+            expect(b.equalTo(c)).toBe(BrsBoolean.False);
+            expect(a.equalTo(d)).toBe(BrsBoolean.True);
+            expect(d.equalTo(a)).toBe(BrsBoolean.True);
+        });
+    });
+
+    test("toString", () => {
+        expect(new roDouble(new Double(22.456)).toString()).toBe("22.456");
+    });
+
+    describe("ifDouble, ifToStr", () => {
+        let a, b, interpreter, someNumberA, someNumberB;
+
+        beforeEach(() => {
+            someNumberA = (Math.random() * 4 + 7) * (Math.random() * 100 + 3);
+            someNumberB = (Math.random() * 4 + 7) * (Math.random() * 100 + 3);
+            a = new roDouble(new Double(0));
```

Comment:
(applies to `RoDouble.test.js`, `RoFloat.test.js`, and `RoInt.test.js`): We're overwriting a and b in two of the three tests within this `describe` – should they just be test-local variables?